scripts/migration/dataset_paths_ecs_rollback.py

```diff
@@ -0,0 +1,259 @@
+
+    # hdfsPublishPath
+    if "hdfsPublishPath" in data_update:
+        hdfs_publish_path = data_update["hdfsPublishPath"]
+        mappings.append(f"hdfsPublishPath: -> {hdfs_publish_path}")
+
+    return ", ".join(mappings)
+
+def rollback_pathchange_entities(target_db: MenasDb, collection_name: str, entity_name: str, entity_names_list: List[str],
+                                 mapping_settings: RollbackSettings, dryrun: bool) -> None:
+
+    assert entity_name == "dataset" or entity_name == "mapping table" , "this method supports datasets and MTs only!"
+
+    if not entity_names_list:
+        print("No {}s to rollback path-changes in {}, skipping.".format(entity_name, collection_name))
+        return
+
+    print("Rollbacking path-change of collection {} started".format(collection_name))
+    dataset_collection = target_db.mongodb[collection_name]
+
+    query =  {"$and": [
+        {"name": {"$in": entity_names_list}}  # dataset/MT name
+    ]}
+
+    docs_count = dataset_collection.count_documents(query)
+    docs = dataset_collection.find(query)
+
+    print("Found: {} {} documents for a potential path change. In progress ... ".format(docs_count, entity_name))
+
+    patched = 0
+    failed_count = 0
+    for item in docs:
+        # item preview
+        if verbose:
+            print("Rollbacking path changes for {} '{}' v{} (_id={}).".format(entity_name, item["name"], item["version"], item["_id"]))
+
+        update_data_set, update_data_unset = update_data_for_item(item, mapping_settings)
+        if update_data_set != {} and update_data_unset != {}:
+            if dryrun:
+                print("  *would rollback* {}".format(data_update_to_nice_string(update_data_set)))
+                print("")
+
+            else:
+                try:
+                    if verbose:
+                        print("  *rollbacking*: {}".format(data_update_to_nice_string(update_data_set)))
+
+                    update_result = dataset_collection.update_one(
+                        {"$and": [
+                            {"_id": item["_id"]}
+                        ]},
+                        {"$set": update_data_set, "$unset": update_data_unset}
+                    )
+                    if update_result.acknowledged and verbose:
+                        print("Successfully rollbacked changed path for {} '{}' v{} (_id={}).".format(entity_name, item["name"], item["version"], item["_id"]))
+                        print("")
+
+                except Exception as e:
+                    print("Warning: Error while change-path rollbacking for {} '{}' v{} (_id={}): {}".format(entity_name, item["name"], item["version"], item["_id"], e))
+                    failed_count += 1
+                else:
+                    patched += 1
+        else:
+            if verbose:
+                print("Nothing left to rollback for {} '{}' v{} (_id={}).".format(entity_name, item["name"], item["version"], item["_id"]))
```

Comment:
I would propose using additional indentation here to link and empty the line below. I needed some time to get the message connected to the previous line. I thought that it was somehow connected to the common process flow.

![image](https://github.com/AbsaOSS/enceladus/assets/109138042/7fd1e071-000e-4ad5-9197-f9cbe5cde1fc)

Reply:
Good idea, applied. Looks like this now:
```
Found: 2 mapping table documents for a potential path change. In progress ...
Rollbacking path changes for mapping table 'SourceSystemMappingTable' v5 (_id=5b6d732ba43a28a6151422aa).
  Nothing left to rollback for mapping table 'SourceSystemMappingTable' v5 (_id=5b6d732ba43a28a6151422aa).

Rollbacking path changes for mapping table 'SourceSystemMappingTable' v1 (_id=5abbaa1e8cdba293c9f0b5a3).
  Nothing left to rollback for mapping table 'SourceSystemMappingTable' v1 (_id=5abbaa1e8cdba293c9f0b5a3).

Successfully rollbacked 0 of 2 mapping table entries, failed: 0
```